x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/InternalTopMetrics.java

```diff
@@ -128,37 +136,38 @@ public boolean isMapped() {
     public XContentBuilder doXContentBody(XContentBuilder builder, Params params) throws IOException {
         builder.startArray("top");
         for (TopMetric top : topMetrics) {
-            top.toXContent(builder, metricName);
+            top.toXContent(builder, metricNames);
         }
         builder.endArray();
         return builder;
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(super.hashCode(), sortOrder, metricName, size, topMetrics);
+        return Objects.hash(super.hashCode(), sortOrder, metricNames, size, topMetrics);
     }
 
     @Override
     public boolean equals(Object obj) {
         if (super.equals(obj) == false) return false;
         InternalTopMetrics other = (InternalTopMetrics) obj;
         return sortOrder.equals(other.sortOrder) &&
-            metricName.equals(other.metricName) &&
+            metricNames.equals(other.metricNames) &&
             size == other.size &&
             topMetrics.equals(other.topMetrics);
     }
 
     @Override
     public double value(String name) {
-        if (metricName.equals(name)) {
-            if (topMetrics.isEmpty()) {
-                return Double.NaN;
-            }
-            assert topMetrics.size() == 1 : "property paths should only resolve against top metrics with size == 1.";
-            return topMetrics.get(0).metricValue;
+        int index = metricNames.indexOf(name);
+        if (index < 0) {
+            throw new IllegalArgumentException("known metric [" + name + "]");            
```

Comment:
+1 - it should be unknown metric. I don't believe we'll *ever* throw it because of the interplay with `hasValue`. Which bothers me, but is a problem for another PR.